neuro_san/client/streaming_input_processor.py

```diff
@@ -89,7 +89,7 @@ def process_once(self, state: Dict[str, Any]) -> Dict[str, Any]:
             # Update the state if there is something to update it with
             chat_context = self.processor.get_chat_context()
             last_chat_response = self.processor.get_answer()
-            returned_sly_data: Dict[str, Any] = self.processor.get_sly_data()
+            sly_data: Dict[str, Any] = self.processor.get_sly_data()
```

Comment:
sly_data coming back might not be the same thing as sly_data going to the agent.
You might want to update() the client's sly_data dictionary instead